### DISCUSSION TITLE
Alert Manager

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#4002] [Feature Requests]
We should add a no-code alert manager for sending emails (and accompanying system notifications when logged in) for specific events. This could be enhanced further once we have a Task Manager (https://github.com/directus/directus/discussions/3062 so it's more performant instead of doing it on every event). This could be enhanced even further once we have Custom Email Templates for the alerts.

This could be within Settings (Admin Only), but I can't think of a reason that anyone shouldn't be able to register for custom alerts. In that case, it could go under the User Profile or even under an Alerts Module.

### Triggers

* System Events
* Data Events
* Aggregate Function Values

### Config

1. Choose an Alert Event, eg: _On Update Articles_
2. Choose Data-Set Filter, eg: _Filter Rules JSON: Published, Author is Me, etc_
3. Choose Value Function, eg: SUM, AVG, MIN, MAX, FIRST, COUNT, etc
4. Choose Conditional, eg: _gte (operator), `100` (value)
5. Choose Alert
    * Email: eg: Subject (Display Template), Body (Display Template), Email Template (Soon)
    * Webhook: eg: What data to send over HTTP request
    * Run Logic: eg: Execute custom logic, such as writing a new record in the API/DB
6. Choose Interval, eg: Check On Event (may not be performant), Check On Interval (Requires Task Manager, eg: `30m`)

### Examples

* Alert me when a new article is ready for me to review
* Alert me if we get any errors (system or from a user's collection)
* Alert me when MRR goes above $100K
* Alert me when a new invoice is created
* Alert me when I'm mentioned in a comment

**Comment by workatease** [comment]
This looks promising

**Reply by benhaynes** [reply]
Don't forget to give it an upvote (the little arrow beside the title). The voting arrows are new to GitHub, but that's how we'll be sorting "most requested" features.

**Reply by workatease** [reply]
done

**Comment by ernst86913** [comment]
Really like the idea.. and would add other channels as well like Push Notification, SMS,... via 3rd party provider integration like PubNub,....

**Comment by joselcvarela** [comment]
With the introduction of Insights module, we are a bit closer to make this happen.

**Comment by Attacler** [comment]
Isnt this like the dataflows that are comming?

**Reply by VaZark** [reply]
I’m assuming the dataflows system evolved from brainstorming for this. The alerts can probably now be managed as an custom action.

It’ll probably need a bit more work if the user is going to be allowed to setup custom flows or opt out of global flows

**Reply by rijkvanzanten** [reply]
That's true! Data flows combined with the notifications system that was already introduced earlier allows for very flexible alerting 🙂

**Reply by sxsws** [reply]
Maybe there should be some default workflows that come with an install of directus for things like this, but with the ability for the user to edit them.

**Reply by benhaynes** [reply]
Absolutely. It would be good to do some polling about what the most useful pre-defined (but inactive) Flows could be included OOTB.

**Comment by rijkvanzanten** [comment]
This is now implemented through Flows!
